File: 2.AddTwoNumbers.cpp
// https://leetcode.com/problems/add-two-numbers/?envType=problem-list-v2&envId=linked-list

ListNode* append(ListNode *node, int value) {
    ListNode* Newnode = new ListNode;
    Newnode->val = value;
    Newnode->next = nullptr;
    if (node == nullptr) {
        return Newnode;
    }
    ListNode* temp = node;
    while (temp->next != nullptr) {
        temp = temp->next;
    }
    temp->next = Newnode;

    return node;
}
class Solution {
public:
    ListNode* addTwoNumbers(ListNode* node1, ListNode* node2) {
        ListNode* ans = nullptr;
        string s1 = "";
        ListNode * temp1 = node1;
        while (temp1 != nullptr) {
            s1 = to_string(temp1->val) + s1;
            temp1 = temp1->next;
        }
        
        ListNode* temp2 = node2;
        string s2 = "";
        while (temp2 != nullptr) {
            s2 = to_string(temp2->val) + s2;
            temp2 = temp2->next;
        }
        
        double x = stod(s1)+ stod(s2);
        string a = to_string(x);
        string s = "";
        for (auto i : a){
            if (isdigit(i)){
                s += i;
            }else{
                break;
            }
        }
        for (int i = s.length() - 1; i >= 0; i--){
            ans =append(ans, s[i] - '0');
        }
        return ans;
    }
};

File: leetcode/2.AddTwoNumbers.py
#  https://leetcode.com/problems/add-two-numbers/?envType=problem-list-v2&envId=linked-list

def ad(node, value):
    new_node = ListNode()
    new_node.val = value
    new_node.next = None
    if node is None:
        return new_node
    temp = node
    while temp.next != None:
        temp = temp.next
    temp.next = new_node
    return node

class Solution:
    def addTwoNumbers(self, node1, node2):
        num1 = 0
        p1 = 0
        temp1 = node1
        while temp1 != None:
            num1 += temp1.val * (10 ** p1)
            p1 += 1
            temp1 = temp1.next
        
        num2 = 0
        p2 = 0
        temp2 = node2
        while temp2 != None:
            num2 += temp2.val * (10 ** p2)
            p2 += 1
            temp2 = temp2.next
        x = num1 + num2
        listi = list(map(int, str(x).replace("", " ").split()))[::-1]
        ans = None
        for i in range(len(listi)):
            ans = ad(ans, listi[i])
        return ans